janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java

```diff
@@ -49,13 +50,14 @@
     private static final Logger log =
             LoggerFactory.getLogger(JanusGraphVertexDeserializer.class);
 
-    public JanusGraphVertexDeserializer(final JanusGraphHadoopSetup setup) {
-        this.setup = setup;
+    public JanusGraphVertexDeserializer(final Configuration config) {
+        this.setup = new JanusGraphHadoopSetup(config);
```

Comment:
Generally it is a bad practice to have `new` operators in class constructors because it violates Dependency Inversion Principle. If we can create a constructor without concrete objects creations in it then it would be a better option.
I.e. the old constructor `public JanusGraphVertexDeserializer(final JanusGraphHadoopSetup setup) {this.setup = setup; ...}` is better then the new one `public JanusGraphVertexDeserializer(final Configuration config) {this.setup = new JanusGraphHadoopSetup(config); ...}`

Reply:
reverted